solr/core/src/java/org/apache/solr/response/transform/ChildDocTransformerFactory.java

```diff
@@ -18,6 +18,7 @@
 
 import static org.apache.solr.schema.IndexSchema.NEST_PATH_FIELD_NAME;
 
+import jakarta.inject.Named;
```

Comment:
To solve the problem of identifying the name for plugins, I found this annotation, which seems quite suitable.  I read further about it when looking at HK2 (a dependency of ours) lightweight dependency-injection framework that uses the annotations in that JAR/package.  This PR doesn't use HK2 BTW.

Reply:
Could also have added a getName to instances but it's more verbose and forces instantiation of the class.  They are instantiated here anyway but nonetheless I like the idea of a future optimization that doesn't instantiate until first use.  Could also decide a public static final String field `NAME` is the name.  Anyway, FWIW I prefer the annotation; feels more modern and is simpler.  Wouldn't be okay in SolrJ (e.g. for DocRouter) as we don't want to add dependencies there but that's fine.